uefi/src/proto/media/load_file.rs

```diff
@@ -0,0 +1,86 @@
+//! Load file support protocols.
+
+#[cfg(feature = "alloc")]
+use alloc::vec::Vec;
+
+use crate::proto::device_path::{FfiDevicePath, DevicePath};
+use crate::proto::unsafe_protocol;
+use crate::{Result, Status};
+use core::ffi::c_void;
+use core::ptr;
+
+/// The UEFI LoadFile2 protocol.
+///
+/// This protocol has a single method to load a file according to some
+/// device path.
+///
+/// This interface is implemented by many devices, e.g. network and filesystems.
+#[derive(Debug)]
+#[repr(C)]
+#[unsafe_protocol("4006c0c1-fcb3-403e-996d-4a6c8724e06d")]
+pub struct LoadFile2Protocol {
+    load_file: unsafe extern "efiapi" fn(
+        this: &mut LoadFile2Protocol,
+        file_path: *const FfiDevicePath,
+        boot_policy: bool,
+        buffer_size: *mut usize,
+        buffer: *mut c_void,
+    ) -> Status,
+}
```

Comment:
We've started putting the protocol definitions into the `uefi-raw` crate, so this could move into `uefi-raw/src/protocol/media.rs`. Then in the `uefi` crate you'll have a `repr(transparent)` wrapper. See https://github.com/rust-osdev/uefi-rs/blob/main/uefi/src/proto/rng.rs#L9-L13 for example.

Reply:
Addressed.